File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        int maxSum = INT_MIN;
        int level = 0;
        int ans = 0;
        while(!q.empty()){
            int size = q.size();
            int currSum = 0;
            while(size--){
            auto node = q.front();
            q.pop();
            currSum += node->val;
            if(node->left)
                q.push(node->left);
            if(node->right)
                q.push(node->right);
            }
            level++;
            if(currSum>maxSum){
                ans = level;
            }
            maxSum = max(maxSum,currSum);
        }
        return ans;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        maxSum = -float('inf')
        level, maxlevel = 0, 0
        q = collections.deque()
        q.append(root)
        while q:
            level += 1
            currSum = 0
            for i in range(len(q)):
                node = q.popleft()
                currSum += node.val
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            if currSum > maxSum:
                maxSum = currSum
                maxlevel = level
        
        return maxlevel